geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloFeatureIndex.scala

```diff
@@ -151,12 +203,17 @@ trait AccumuloWritableIndex extends AccumuloFeatureIndexType {
     }
   }
 
-  /**
-    * Indicates whether the ID for each feature is serialized with the feature or in the row
-    *
-    * @return
-    */
-  def serializedWithId: Boolean
+  // back compatibility check for old metadata keys
+  abstract override def getTableName(typeName: String, ds: AccumuloDataStore): String = {
+    lazy val oldKey = this match {
+      case i if i.name == RecordIndex.name    => "tables.record.name"
+      case i if i.name == AttributeIndex.name => "tables.idx.attr.name"
+      case i => s"tables.${i.name}.name"
+    }
+    Try(super.getTableName(typeName, ds)).recoverWith {
```

Comment:
Idly I wonder if it'd be better to move 'oldKey' into the recoverWith and make it a val?